crates/spirv-std/src/arch.rs

```diff
@@ -5,6 +5,72 @@
 //! no additional safety checks beyond type-checking.
 use crate::{scalar::Scalar, vector::Vector};
 
+/// Result is true if any component of `vector` is true, otherwise result is
+/// false.
+#[spirv_std_macros::gpu_only]
+#[doc(alias = "OpAny")]
+#[inline]
+pub fn any<V: Vector<bool, N>, const N: usize>(vector: V) -> bool {
+    let mut result = false;
+
+    unsafe {
+        asm! {
+            // Types & Constants
+            "%bool = OpTypeBool",
+            "%uchar = OpTypeInt 8 0",
+            "%uchar_0 = OpConstant %uchar 0",
+            "%uchar_1 = OpConstant %uchar 1",
+            "%glam_vec_type = OpTypeVector %uchar {len}",
+            "%bool_vec_type = OpTypeVector %bool {len}",
+            "%false_vec = OpConstantNull %glam_vec_type",
+            // Code
+            "%vector = OpLoad %glam_vec_type {vector}",
+            "%bool_vec = OpINotEqual %bool_vec_type %vector %false_vec",
+            "%result = OpAny %bool %bool_vec",
+            "%boolean = OpSelect %uchar %result %uchar_1 %uchar_0",
+            "OpStore {result} %boolean",
+            vector = in(reg) &vector,
+            len = const N,
```

Comment:
Love that parameterizing inline `asm!` by `const` generics is a thing now! (I wonder how it's implemented but I'm getting distracted...)